src/brevitas/core/scaling/runtime.py

```diff
@@ -148,54 +132,59 @@ def __init__(
             dtype,
             device)
         self.stats_scaling_impl = _StatsScaling(
-            restrict_scaling_impl,
-            restrict_threshold_impl,
-            scaling_shape,
-            scaling_min_val,
-            affine_rescaling,
-            affine_shift_scale,
-            dtype,
-            device)
+            restrict_scaling_impl=restrict_scaling_impl,
+            restrict_threshold_impl=restrict_threshold_impl,
+            affine_rescaling_impl=affine_rescaling_impl,
+            scaling_min_val=scaling_min_val)
 
     @brevitas.jit.script_method
     def forward(self, x: torch.Tensor, threshold: Optional[torch.Tensor] = None) -> torch.Tensor:
         stats = self.runtime_stats(x)
         return self.stats_scaling_impl(stats, threshold)
 
 
-class _AffineRescaling(brevitas.jit.ScriptModule):
+class ParameterAffineRescaling(brevitas.jit.ScriptModule):
 
-    def __init__(
-            self,
-            scaling_shape,
-            shift_scale,
-            dtype: Optional[torch.dtype],
-            device: Optional[torch.device]):
-        super(_AffineRescaling, self).__init__()
-        self.affine_weight = Parameter(torch.ones(scaling_shape, dtype=dtype, device=device))
-        if shift_scale:
-            self.affine_bias = ParameterWrapper(
-                torch.zeros(scaling_shape, dtype=dtype, device=device))
-        else:
-            self.affine_bias = StatelessBuffer(torch.tensor(0., dtype=dtype, device=device))
+    def __init__(self, scaling_shape, dtype: Optional[torch.dtype], device: Optional[torch.device]):
+        super(ParameterAffineRescaling, self).__init__()
+        self.value = Parameter(torch.ones(scaling_shape, dtype=dtype, device=device))
 
     @brevitas.jit.script_method
     def forward(self, x):
-        out = x * self.affine_weight + self.affine_bias()
+        out = x * self.value
         out = abs_binary_sign_grad(out)
         return out
 
     def _load_from_state_dict(
             self, state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys,
             error_msgs):
-        super(_AffineRescaling, self)._load_from_state_dict(
+        super(ParameterAffineRescaling, self)._load_from_state_dict(
             state_dict, prefix, local_metadata, strict, missing_keys, unexpected_keys, error_msgs)
-        affine_weight_key = prefix + 'affine_weight'
-        affine_bias_key = prefix + 'affine_bias'
-        if config.IGNORE_MISSING_KEYS and affine_weight_key in missing_keys:
-            missing_keys.remove(affine_weight_key)
-        if config.IGNORE_MISSING_KEYS and affine_bias_key in missing_keys:
-            missing_keys.remove(affine_bias_key)
+        value_key = prefix + 'value'
+        if config.IGNORE_MISSING_KEYS and value_key in missing_keys:
+            missing_keys.remove(value_key)
+
+
+class ConstAffineRescaling(brevitas.jit.ScriptModule):
```

Comment:
If we pass `affine_rescaling_init` to `_AffineRescaling`, then we do not need this class

Reply:
This has other disadvantages compared to `_AffineRescaling`. For example, since you saved it as a stateless buffer, it won't be picked up in the state dict, which means that saving/loading a checkpoint with this rescaling will have unexpected behaviours